var/spack/repos/builtin/packages/apptainer/package.py

```diff
@@ -46,3 +46,18 @@ class Apptainer(SingularityBase):
     @property
     def config_options(self):
         return []
+
+    # Hijack the edit stage to run mconfig.
+    def edit(self, spec, prefix):
+        with working_dir(self.build_directory):
+            confstring = "./mconfig --prefix=%s" % prefix
+            if "~suid" in spec:
+                confstring += " --without-suid"
```

Comment:
Can you change this upstream, in `SingularityBase`, to be explicit on the option?

Reply:
that configure argument didn't exist until apptainer 1.1.0, so I don't think it would work for singularity.

Reply:
Actually that's not true. I can make that change

Reply:
It's already there: https://github.com/spack/spack/blob/42a452d54c8a25f9f415fef8cf9e3a5c64b7a46a/var/spack/repos/builtin/packages/singularityce/package.py#L84

Reply:
Because I'm overwriting the whole `edit` method, I copy pasted what's already there and only added this:
```python
            if "+suid" in spec and spec.satisfies("@1.1.0:"):
                confstring += " --with-suid"
```

Reply:
Can't you just override `self.config_options`, instead of duplicating `edit`?

Reply:
How can I make sure that my override happens after `edit` happens?